### PR TITLE
Various fixes and database improvements

### PR DESCRIPTION
**Pull request by Dovias**
I think we should merge db-improvements branch with main branch, since it looks good to go for further development. Please review it, of course 😄 

1. Fixed a typo with the project's name. **"insurence" -> "insurance"**. This typo fix affects the whole package path, so take in note this, when pulling changes from the branch. You might need to move some things that are not on the main branch to the new package name. I think its not too late to change this since we only did some initialization of the project.

2. Added all the necessary SQL initialization scripts with proper liquibase SQL changeset rules.

3. Moved the h2 console to the /admin/db subpath, and the path is now not hardcoded inside the code anymore. As implied by SecurityConfig configuration component, database console can now be accessed with admin role only. You can change the path in spring boot's application.yaml file.

4. Fixed the bug with the h2 console which did not permit users to use the h2 console after they've inputted the proper credentials.

5. Initialized the default database schema inside separate database_schema changeset. There's no need to specify the schema inside every changeset and java entity classes anymore, since it is the first changeset that gets runned.

### CODE REVIEW

**Review by BlenderEngineer — DISMISSED** [review]

**Review by NorthDragonDuck — APPROVED** [review]

**Comment by Dovias** [comment]
Added the final change for this branch regarding the upcoming access privileges to the backend api. Please re-review it

**Review by NorthDragonDuck — APPROVED** [review]

**Review by BlenderEngineer — APPROVED** [review]

File: template/src/main/resources/db/changelog/changes/user_table.sql
Comment (anton-kolyaev) [inline template/src/main/resources/db/changelog/changes/user_table.sql:1]: you should not alter ANY already merged changelogs - that will lead to fail validation of checksum on real database. Only incremental updates.
Rollback that please.

**Review by anton-kolyaev — CHANGES_REQUESTED** [review]
Rephrase your commit message to pattern like:
`[POT-1]  Enable user to login`
https://opensource.com/article/22/12/git-commit-message

Split your PR into logical pieces. I'm quite sure that table 'companies' relates to another ticket and even to another person.

**Comment by Dovias** [comment]
This pull request is broken and should not be used in production (especially the database structure, since it could lead to circular dependency issues due to ordering of execution of rollbacks in Liquibase).  Correct changes were transferred to auth-impl branch.

**Pull request closed without merging**